scripts/process.py

```diff
@@ -0,0 +1,161 @@
+        for field_name in field:
+            rows[row_number].append(data_dict[iex_type][field_name[0]])
+
+
+def extract_financial_data():
+    # get constituents data
+    reader = csv.reader(open('data/constituents.csv'))
+
+    # gather field titles and IEX field names
+    iex_fieldnames = []
+    field_titles = []
+    for iex_type in sorted(IEX_TYPES_FIELDNAMES_TITLES.values()):
+        iex_fieldnames += [field[0] for field in iex_type]
+        field_titles += [field[1] for field in iex_type]
+
+    outrows = [row for row in reader]
+    # add new column titles to outfile
+    outrows[0] += field_titles + ['SEC Filings']
+    # gather symbols
+    symbols = [row[0] for row in outrows[1:]]
+    # gather IEX "types"
+    iex_types = IEX_TYPES_FIELDNAMES_TITLES.keys()
+    # build first part of url, IEX uses comma delimiters between values
+    url = IEX_BASE_URL + '?'
+    url += 'types=' + ','.join(iex_types)
+    url += '&filter=' + ','.join(iex_fieldnames)
+    url += '&symbols='
+
+    # make requests in chunks of symbols at a time
+    # store response data in outrows
+    for index in range(0, len(symbols), IEX_NUMBER_OF_SYMBOLS_PER_REQUEST):
+        query = url + ','.join(symbols[index:index + IEX_NUMBER_OF_SYMBOLS_PER_REQUEST])
+        resp = urllib.request.urlopen(query)
+        # use ordered dict to keep results sorted in original order
+        results = (json.loads(resp.readline().decode('utf-8'), object_pairs_hook=OrderedDict))
+        # loop over chunk results and store respective fields into outrows
+        for count, stock in enumerate(results):
+            outrows_index = index + count + 1
+            # make sure we're updating the correct row
+            assert(outrows[outrows_index][0] == stock)
+            # append column data in order of IEX types and field_names
+            append_column_data(outrows, outrows_index, results[stock])
+            # append edgar url
+            outrows[outrows_index].append(EDGAR_BASE_URL + stock)
+            yield {
+                'Symbol': outrows[outrows_index][0],
+                'Name': outrows[outrows_index][1],
+                'Sector': outrows[outrows_index][2],
+                'Price': outrows[outrows_index][3],
+                'Price/Earnings': outrows[outrows_index][4],
+                'Dividend Yield': outrows[outrows_index][5],
+                'Earnings/Share': outrows[outrows_index][6],
+                '52 Week Low': outrows[outrows_index][7],
+                '52 Week High': outrows[outrows_index][8],
+                'Market Cap': outrows[outrows_index][0],
+                'EBITDA': outrows[outrows_index][10],
+                'Price/Sales': outrows[outrows_index][11],
+                'Price/Book': outrows[outrows_index][12],
+                'SEC Filings': outrows[outrows_index][13]
+            }
+
+
+with urllib.request.urlopen("https://en.wikipedia.org/wiki/List_of_S%26P_500_companies") as response:
```

Comment:
Think it's better to send a request from inside of the `extract_constituents()` instead of doing so. What's the use of `with ... as resp: ...`

Reply:
>`What's the use of with ... as resp: ...`

It's the same as response = ..., html = response.read()

>Think it's better to send a request from inside of the extract_constituents() instead of doing so.

No, the script is not structured in that way for that to work. However I might be able to load constituents from a different source. That might be a more elegant solution